aqt/installer.py

```diff
@@ -646,12 +706,68 @@ def _create_parser(self):
         )
         self._set_common_options(tools_parser)
         #
-        list_parser = subparsers.add_parser("list")
+        # list_parser = subparsers.add_parser("list")
+        # list_parser.set_defaults(func=self.run_list)
+        # self._set_common_argument(list_parser)
+        # help_parser = subparsers.add_parser("help")
+        # help_parser.set_defaults(func=self.show_help)
+        # parser.set_defaults(func=self.show_help)
+
+        # list2: placeholder name for added functionality; don't know what to call it!
+        list_parser = subparsers.add_parser(
+            "list",
+            formatter_class=argparse.RawTextHelpFormatter,
+            epilog="Examples:\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9                   # print all versions of Qt 5.9\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9 --latest-version  # print latest Qt 5.9\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9 --latest-packages # print packages for latest 5.9\n"
+            "$ aqt list qt5 windows desktop --list-packages 5.12.0             # print packages for 5.12.0\n",
+        )
         list_parser.set_defaults(func=self.run_list)
-        self._set_common_argument(list_parser)
-        help_parser = subparsers.add_parser("help")
-        help_parser.set_defaults(func=self.show_help)
-        parser.set_defaults(func=self.show_help)
+        list_parser.add_argument(
+            "category",
+            choices=["tools", "qt5", "qt6"],
+            help="category of packages to list",
+        )
+        list_parser.add_argument(
+            "host", choices=["linux", "mac", "windows"], help="host os name"
+        )
+        list_parser.add_argument(
+            "target", choices=["desktop", "winrt", "android", "ios"], help="target sdk"
+        )
+        list_parser.add_argument(
+            "--extension",
+            choices=ALL_EXTENSIONS,
+            help="extension of packages to list",
+        )
+        # list_parser.add_argument("qt_version", required=False, type=str,
+        #                          help='Qt version in the format of "5.X.Y". If this is set, then ')
+        list_parser.add_argument(
+            "--filter-minor",
+            type=int,
+            metavar="MINOR_VERSION",
+            help="print versions for a particular minor version. "
+            "IE: `aqt list qt5 windows desktop --filter-minor 12` prints all versions beginning with 5.12",
+        )
+        output_modifier_exclusive_group = list_parser.add_mutually_exclusive_group()
+        output_modifier_exclusive_group.add_argument(
+            "--list-packages",
+            type=str,
+            metavar="VERSION",
+            help='Qt version in the format of "5.X.Y". '
+            "When set, this lists all the packages available for Qt 5.X.Y.",
+        )
```

Comment:
> When user specify `category host target version` then list packages by reading Updates.xml.

@miurahr, this is the optional argument I am using to list packages from Updates.xml. Personally, I prefer to require a flag in front of optional arguments like this because I think it makes the interface easier to understand. If you disagree with this choice, please let me know and I will change it to what you originally asked for.

Reply:
It looks better than my proposal. Please feel free to change UI because existent  list sub command is quite incomplete.